lib/bundler/definition.rb

```diff
@@ -409,8 +409,10 @@ def ensure_equivalent_gemfile_and_lockfile(explicit_flag = false)
              "updated #{Bundler.default_lockfile.relative_path_from(SharedHelpers.pwd)} to version control."
 
       unless explicit_flag
+
+        suggest_command = Bundler.settings.locations("frozen")[:global] == "1" ? "bundle config --delete frozen" : "bundle install --no-deployment"
```

Comment:
IIRC, deleting both was our way of punting on the issue. It seems like terrible UI to require specific combinations of multiple flags. So far, it has worked. 😬  We might need to change it if there is an actual usecase that requires deleting one and not the other, though.